### PR TITLE
Group7

### PR DESCRIPTION
**Pull request by urshala**
Group 7
Deepak
Delio Castillo

### CODE REVIEW

File: calculator/main.py [hunk 1]
```diff
@@ -26,8 +27,23 @@ def perform_operation(calc, operation, params):
...
-
+    current_operation = calc['operations'].get(operation, None)
+    if not current_operation:
+        raise InvalidOperation('Given operation is invalid.')
+    
+    try:
+        [float(x) for x in params]
+    except ValueError:
+        raise InvalidParams('Given params are invalid.')
+
```
Comment (koakekuna) [inline calculator/main.py:38]: I like this try block, very nice!

File: calculator/main.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 from calculator.operations import *
 from calculator.exceptions import *
-
+import re
```
Comment (koakekuna) [inline calculator/main.py:6]: didn't see any regexes?

File: calculator/main.py [hunk 3]
```diff
@@ -26,8 +27,23 @@ def perform_operation(calc, operation, params):
...
+        [float(x) for x in params]
+    except ValueError:
+        raise InvalidParams('Given params are invalid.')
+        
+    # perform the calculation and store the result
+    result = current_operation(*params)
+     
+    # add to the history of the calculator
+    calc['history'].append((str(datetime.now()), operation, params, result))
+
```
Comment (koakekuna) [inline calculator/main.py:44]: excellent! exactly how we approached it too

File: calculator/operations.py [hunk 4]
```diff
@@ -1,23 +1,34 @@
...
 def divide(*args):
-    # your implementation here
-    pass
+    #  Returns the division of numbers
+    total = float(args[0])
+    print (total)
+    for i in args[1:]:
+        total = total / i
+        print(total, i)
+    return total
```
Comment (anshulxyz) [inline calculator/operations.py:23]: You can approach the `divide()` the same way you did with `subtract()` using hint from your `multiply()`:

``` python
return args[0] / reduce(lambda x,y: x*y, args[1:], 1)
```

**Pull request closed without merging**